compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -451,6 +451,9 @@ fn collect_items_rec<'tcx>(
                                     collect_const_value(tcx, val, &mut neighbors);
```

Comment:
Since `collect_const_value` isn't doing its job here, you can just remove it entirely.